### PR TITLE
add matrix for dns type vs kind

### PR DESCRIPTION
**Pull request by dhiaayachi**
This PR add a matrix to document the current DNS server behaviour per DNS query Type vs dns query kind

### CODE REVIEW

**Comment by jsosulska** [comment]
Hi @dhiaayachi - Is this something that we may want to add to our Docs site, instead of just the codebase?

File: contributing/service-discovery/dns-matrix.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+|                | service                 | connect (enterprise)    | ingress (enterprise)    | node                   | query           | addr                         |
```
Comment (dnephin) [inline contributing/service-discovery/dns-matrix.md:1]: A quick intro paragraph would be great to explain what this is, and maybe also to say that only a single `OpCode` is supported.

**Review by dnephin — COMMENTED** [review]
Nice! Do you think it would make sense to include on the same page as the docs in #10440 ?

**Comment by dnephin** [comment]
>  Is this something that we may want to add to our Docs site, instead of just the codebase?

I think our existing docs for the [DNS interface](https://www.consul.io/docs/discovery/dns) are probably in a better format for users. I would expect a table like this to mostly only be useful for someone looking to understand the code.

Ideally the code would be structured in a way that makes the matrix obvious, but until then the docs will help.

**Comment by dhiaayachi** [comment]
> Nice! Do you think it would make sense to include on the same page as the docs in #10440 ?

Yes If OK I can bring it to the same PR

**Comment by dhiaayachi** [comment]
> Hi @dhiaayachi - Is this something that we may want to add to our Docs site, instead of just the codebase? 

IMO it will be a high maintenance task to keep it in sync with the code specially in the current state. I would wait until the code is properly refactored and reevaluate after if still needed.

**Comment by dhiaayachi** [comment]
I added the change in this PR in #10440

**Pull request closed without merging**